src/Bundler/EntryPoint.php

```diff
@@ -52,58 +64,19 @@ public function getFile(): File
     }
 
     /**
-     * Return all non-vendor files.
-     *
-     * @return DependencyNodeInterface[]
-     */
-    public function getBundleFiles(): array
-    {
-        return $this->bundle_files;
-    }
-
-    /**
-     * Return all vendor files.
+     * Returns the files to build
      *
-     * @return DependencyNodeInterface[]
-     */
-    public function getVendorFiles(): array
-    {
-        return $this->vendor_files;
-    }
-
-    /**
-     * Return all assets.
-     *
-     * @return File[]
-     */
-    public function getAssetFiles(): array
-    {
-        return $this->asset_files;
-    }
-
-    /**
-     * Bundle files are located in the application itself.
-     *
-     * @param string $output_dir
-     * @return File
-     */
-    public function getBundleFile(string $output_dir): File
-    {
-        return new File($output_dir . '/' . sprintf(
-            '%s.bundle.%s',
-            $this->file->getBaseName(),
-            'js'
-        ));
-    }
-
-    /**
-     * Vendor files are third party files, usually located in node_modules.
-     *
-     * @param string $output_dir
-     * @return File
+     * @return array
      */
-    public function getVendorFile(string $output_dir): File
+    public function getFilesToBuild(string $output_dir): array
     {
-        return new File($output_dir . '/' . sprintf('%s.vendor.js', $this->file->getBaseName()));
+        $result = [];
+        foreach ($this->files as $split_file => $dependencies) {
+            $result[$output_dir . '/' . $split_file] = $dependencies;
+        }
+        foreach ($this->asset_files as $asset_file) {
+            $result[$output_dir . '/' . $asset_file->getFile()->getName()] = [$asset_file];
```

Comment:
`getFile()` can also return a `Module`. These do not have an extension in the name. Maybe use `$asset_file->getFile()->path`.